src/SqlAsyncCollector.cs

```diff
@@ -237,22 +252,41 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
         /// to upsert to and returns the extra property names in a List.
         /// </summary>
         /// <param name="columns"> The columns of the table to upsert to </param>
+        /// <param name="row"> First row to get the column names when T is JObject </param>
         /// <returns>List of property names that don't exist in the table</returns>
-        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns)
+        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns, T row)
```

Comment:
Shouldn't this be `T item` or something? A row implies a row in a data table or something, but T is the type of object we're currently handling if I'm understanding this correctly.

Reply:
since we're using item for the input, I thought it would make sense to have it as row, since T is actually the object representation of Table.

Reply:
At the very least something like `rowItem` would be clearer I think. A row has well-known meanings and types, which this isn't.